coralproject/talk
# Generate and integrate introspection query results

## What does this PR do?

- Added new WEBPACK env var which is enabled during yarn build scripts
- Defered redis starting until listen is called
- Moved pubsub to a factory pattern init
- Async/Await'ed the routes
- Moved pubsub handle for routes into middleware
- Adjusted redis cache and job processors to have lazy connection
starting
- Disabled mongo from auto-connecting on require
- Adjusted package redis clients to act as factory singletons instead
- Generate a `introspection.json` as part of our build process.
- Auto update `introspection.json` when using `yarn dev-start`.
- Port client code to use the `IntrospectionFragmentMatcher`.

## How do I test this PR?

1. Stop mongo + redis docker containers.
2. Create a file `test_772.js` with the contents:

```js
require('./app');
```

3. Run `WEBPACK=true NODE_ENV=test node test_772.js`
4. See that the script starts/stops without issue.
